plugins/Marketplace/Plugins.php

```diff
@@ -444,4 +449,18 @@ private function prettifyNumberOfDownloads(&$plugin): void
 
         $plugin['numDownloadsPretty'] = $nice;
     }
+
+    private function getConsumerWisePluginStatus(): array
```

Comment:
The method name is a bit confusing for me. We are getting license status for plugins the consumer has licenses for. Can we do it similarly to how `addMissingRequirements` is used? I'd create a method `addConsumerLicenseStatus($plugin)` and in that method I'd get all the license statuses (using a private variable to cache the call) and then just assign the license status based on the plugin name directly to the plugin metadata.

Reply:
@michalkleiner refactored the code, can you check once ?